src/main/java/org/cryptomator/common/vaults/Vault.java

```diff
@@ -142,12 +165,22 @@ public synchronized void unlock(MasterkeyLoader keyLoader) throws CryptoExceptio
 		if (cryptoFileSystem.get() != null) {
 			throw new IllegalStateException("Already unlocked.");
 		}
+		var fallbackProvider = mountProviders.stream().findFirst().orElse(null);
+		var defMntServ = mountProviders.stream().filter(s -> s.getClass().getName().equals(settings.mountService.getValue())).findFirst().orElse(fallbackProvider);
+		var selMntServ = mountProviders.stream().filter(s -> s.getClass().getName().equals(vaultSettings.mountService.getValue())).findFirst().orElse(defMntServ);
```

Comment:
You already have the `actualMountService` object which incorporates this logic. Hence, you should use it instead of duplicating code. Additionally, i think this logic is better placed inside the Mounter, since it is highly mount related.